### PR TITLE
Start the auction

### PR DESCRIPTION
**Pull request by marpuz**
- feature: starting new auction by Modal in profile view
image

### CODE REVIEW

File: frontend/src/API/UserService/auctions.ts [hunk 1]
```diff
@@ -2,9 +2,23 @@ import { AxiosResponse } from "axios";
 import { AuctionItem } from "interfaces";
 import axiosInstance from "services/axios";
 
+interface item123 {
```
Comment (kkolasinski) [inline frontend/src/API/UserService/auctions.ts]: what happened to item122? 👀

File: frontend/src/components/Modal/AddAuctionModal/AddAuctionModal.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            setResponse(null);
+        }, 2000);
+    };
+
+    const formik: FormikValues = useFormik({
+        initialValues: init,
+        validationSchema,
+        validateOnChange: false,
+        onSubmit: async (values) => {
+            console.log(values);
```
Comment (kkolasinski) [inline frontend/src/components/Modal/AddAuctionModal/AddAuctionModal.tsx]: 👀

File: frontend/src/components/createdProduct/CreatedProduct.tsx [hunk 3]
```diff
@@ -19,32 +24,52 @@ export const CreatedProduct = ({ item, profileID }: CreatedProductProps) => {
...
-                                className={styles.deleteButton}
-                                type="button"
-                                onClick={handleDelete}
-                            >
-                                Delete
-                            </button>
-                            <div className={styles.startAuction}>
-                                <button type="button" className={styles.startAuctionButton}>
-                                    <span>Start auction!</span>
+                {activeProfile === item.ownerID &&
```
Comment (kkolasinski) [inline frontend/src/components/createdProduct/CreatedProduct.tsx]: suggestion: move this expression into a variable for readability, could be called something like `isOwner`

File: frontend/src/components/productCard/ProductCard.tsx [hunk 4]
```diff
@@ -44,18 +44,27 @@ export const ProductCard = ({ item }: ProductCardProps) => {
                                 </button>
                             </div>
                             <div className={styles.placeBidContainer}>
-                                <button type="button" className={styles.placeBidButton}>
-                                    <span>Place a bid</span>
-                                </button>
+                                {item.instantSellPrice === true && (
```
Comment (kkolasinski) [inline frontend/src/components/productCard/ProductCard.tsx]: tip: you don't have to explicitly compare to `true` unless the value of `instantSellPrice` is not a boolean (just checking if the value is truthy should be enough in this case - `item.instantSellPrice && ()`)

**Review by kkolasinski — APPROVED** [review]

File: frontend/src/components/Modal/AddAuctionModal/AddAuctionModal.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                <RenderInput
+                    key={item.id}
+                    item={item}
+                    onInputChange={formik.handleChange}
+                    value={formik.values[item.name]}
+                    error={formik.errors[item.id]}
+                />
+            ))}
+            <ToggleInputs onToggleChange={formik.handleChange} />
+            {formik.values["instantSellPrice"] && (
```
Comment (dorotanowak00) [inline frontend/src/components/Modal/AddAuctionModal/AddAuctionModal.tsx]: You can use: formik.values.instantSellPrice instead

**Pull request merged**